bftengine/src/bcstatetransfer/BCStateTran.hpp

```diff
@@ -425,12 +427,21 @@ class BCStateTran : public IStateTransfer {
     FetchingState& fetchingState_;
   };
 
+  // Must be less than config_.refreshTimerMs
+  const uint32_t processCommitsTimeoutMilli_ = 5;
+
   BlockIODataPool ioPool_;
   std::deque<BlockIOContextPtr> ioContexts_;
+  bool oneShotTimerFlag_;  // used to control the trigger of oneShotTimer self requests
 
   // returns number of jobs pushed to queue
   uint16_t getBlocksConcurrentAsync(uint64_t nextBlockId, uint64_t firstRequiredBlock, uint16_t numBlocks);
 
+  // lastBlock: is true if we put the oldest block (firstRequiredBlock)
+  // breakIfNotReady: is true if caller would like to exit if the next future is not ready (job not ended yet).
+  //    In that case, a ONESHOT timer is invoked to check the future again soon.
+  // return: true if done procesing all futures, and false if the front one was not std::future_status::ready
+  bool processCommitResultsAsync(bool lastBlock, bool breakIfNotReady);
```

Comment:
breakIfNotReady is not a clear parameter and complicates both implementation and understanding

Reply:
I will change the name..
The whole idea of putting blocks concurrently relay on this flag.
If I don't break immediately, I'll wait for each block and blocks won't be put concurrently.
If you mean that the name should be renamed - I can change the name, but the flag is a must for the implementation of concurrent putBlock.
Many times, the future for the job we just invoked will not be ready, the main thread will break out and go to the handoff queue to fetch the next ItemDataMsag.

Reply:
new name:
waitForAllJobsToComplete
This has the opposite meaning: when true - cannot break out the loop, must wait for each future in the queue to complete.